### ISSUE TITLE
Stack overflow when attempting to display error types.

### ISSUE THREAD

**Issue by mobile-bungalow** [mikedilger/mime-multipart#15]
It appears that your `Display` implementation calls itself recursively forever.
    
reproduce with: 

```Rust  use crate::error::Error;
     #[test]
     fn display_error() {
         let custom_error = std::io::Error::new(std::io::ErrorKind::Other, "oh no!");
         println!("{}", Error::Io(custom_error));
     }

**Comment by mikedilger** [comment]
Well that's embarrassing. I'll take a look.

**Comment by mikedilger** [comment]
Fixed in f2cba9e1c93c9296e6115b8633ff1dcea69098c5

**Comment by djoyner** [comment]
This is a critical fix. Any chance it could be merged into the `0.6` branch and released as `0.6.2`?

**Comment by mikedilger** [comment]
This was merged and released in `0.6.1`
